search/typo_suggest_engine.go

```diff
@@ -0,0 +1,90 @@
+package search
+
+import (
+	"context"
+	"strconv"
+	"time"
+
+	"github.com/Bnei-Baruch/archive-backend/consts"
+	"github.com/Bnei-Baruch/archive-backend/es"
+	"github.com/pkg/errors"
+	"gopkg.in/olivere/elastic.v6"
+	null "gopkg.in/volatiletech/null.v6"
+)
+
+func (e *ESEngine) GetTypoSuggest(query Query) (null.String, error) {
+	srv := e.esc.Search()
+	suggestText := null.String{"", false}
+
+	if _, err := strconv.Atoi(query.Term); err == nil {
+		//  ignore numbers
+		return suggestText, nil
+	}
+
+	var hasHebrew bool
+	var hasRussian bool
+
+	indices := make([]string, len(query.LanguageOrder))
+	for i := range query.LanguageOrder {
+		if query.LanguageOrder[i] == consts.LANG_HEBREW {
+			hasHebrew = true
+		} else if query.LanguageOrder[i] == consts.LANG_RUSSIAN {
+			hasRussian = true
+		}
+		indices[i] = es.IndexNameForServing("prod", consts.ES_RESULTS_INDEX, query.LanguageOrder[i])
+	}
+	srv.Index(indices...)
+
+	suggestorField := "content.language"
```

Comment:
What is the following language/field strategy?
Can you please explain?

Reply:
Logic seems complex, would it not be simpler to have each case explicitly enumerated?